src/utils/parsers.js

```diff
@@ -0,0 +1,27 @@
+// 커스텀 구분자 추출
+export function extractCustomDelimiter(input) {
+  const CUSTOM_DELIMITER_REGEX = /^\/\/(.+?)\\n/;
+  const customDelimiterMatch = input.match(CUSTOM_DELIMITER_REGEX);
+  return customDelimiterMatch ? customDelimiterMatch[1] : null;
+}
+
+// 숫자 문자열 추출
+export function getNumberString(input, customDelimiter) {
+  const CUSTOM_DELIMITER_REGEX = /^\/\/(.+?)\\n/;
+  const customDelimiterLength = customDelimiter
+    ? input.match(CUSTOM_DELIMITER_REGEX)[0].length
+    : 0;
+  return input.slice(customDelimiterLength).trim();
+}
+
+// 구분자로 문자열을 나누어 숫자 배열로 변환
+export function parseNumbers(input, customDelimiter) {
+  const DEFAULT_DELIMITERS = /[,:]/g;
+  const delimiter = customDelimiter || DEFAULT_DELIMITERS;
+
+  return input
+    .split(delimiter)
+    .map((num) => num.trim())
+    .filter((num) => num !== "")
+    .map(Number);
+}
```

Comment:
이 부분은 한 함수에 여러 기능이 들어있는 것 같아서 분리가 필요하지 않을까 싶어요 아래 방법은 어떨까요

```suggestion

// 구분자로 문자열을 나누는 함수
function splitByDelimiter(input, customDelimiter) {
  const DEFAULT_DELIMITERS = /[,:]/g;
  const delimiter = customDelimiter || DEFAULT_DELIMITERS;

  return input.split(delimiter).map((str) => str.trim());
}

// 문자열 배열을 숫자 배열로 변환하는 함수
function convertToNumbers(stringArray) {
  return stringArray
    .filter((str) => str !== "")  // 빈 문자열 제거
    .map(Number);                 // 숫자로 변환
}

// 구분자로 문자열을 나누어 숫자 배열로 변환하는 함수
export function parseNumbers(input, customDelimiter) {
  const stringArray = splitByDelimiter(input, customDelimiter);
  return convertToNumbers(stringArray);
}
```